deps.sh

```diff
@@ -1,13 +1,36 @@
 #!/bin/bash
 
+# test required libraries
+libs=(libssl-dev libcurl3-dev liblzma-dev libeigen3-dev  \
+    libgoogle-glog-dev libatlas-base-dev libsuitesparse-dev libgflags-dev \
+    libx264-dev libopenjpeg-dev libxvidcore-dev \
+    libpng-dev libjpeg-dev libbz2-dev git python-pip wget \
+    libleveldb-dev libsnappy-dev libhdf5-serial-dev liblmdb-dev python-dev \
+    llvm clang python-tk autoconf autogen libtool libtbb-dev libopenblas-dev \
+    liblapacke-dev swig yasm python2.7 cpio cmake zip)
+
+for lib in "${libs[@]}"
+do
+    dpkg-query -l $lib &> /dev/null
+    if [[ $? != 0 ]]; then
+        echo "$lib is not installed. abort"
+        exit 1
+    fi
+done
```

Comment:
One issue with this check is that a user might have installed some of these dependencies without using the package manager. Having this hard failure here would cause an issue in that case. Perhaps change this to print a  warning instead?

Reply:
Yes I agree. Maybe we can seperate the library check on dependency basis? For instance, we only check `libjpeg-dev` etc if the user wants to install OpenCV?